src/scripts/install_machine.py

```diff
@@ -0,0 +1,334 @@
+            backing_store = f",backing_store={args.get('source')},"
+        else:
+            backing_store = ""
+
+        if args.get('sourceType') == 'disk_image':
+            params.append(f"{args.get('source')},device=disk")
+        elif args.get('storagePool') == 'NoStorage':
+            params.append("none")
+        elif args.get('storagePool') != 'NewVolume':
+            params.append(f"vol={args.get('storagePool')}/{args.get('storageVolume')}{backing_store}")
+        else:
+            params.append(f"size={args.get('storageSize')},format=qcow2{backing_store}")
+
+    # NICs
+    if args.get('type') == "install":
+        has_param = False
+        for network in args.get('vnics'):
+            params += ['--network', network]
+            has_param = True
+
+        if not has_param:
+            params += ['--network', "none"]
+
+    # Consoles
+    if args.get('type') == "install":
+        has_param = False
+        for graphics in args.get('graphics'):
+            params += ['--graphics', graphics]
+            has_param = True
+
+        if not has_param:
+            params += ['--graphics', "none"]
+    else:
+        has_param = False
+        for graphics in get_graphics_capabilies(args.get('connectionName')):
+            params += ['--graphics', f"{graphics},listen=127.0.0.1"]
+            has_param = True
+
+        if not has_param:
+            params += ['--graphics', "none"]
+
+    # Installation media
+    if not args.get('sourceType') == "cloud":
+        prepare_installation_source(params, args)
+
+    # Autostart
+    if args.get('autostart'):
+        params.append("--autostart")
+
+    # VCPUs
+    if args.get('vcpu'):
+        params += ['--vcpus', args.get('vcpu')]
+
+    # Firmware
+    if args.get('firmware'):
+        params += ['--boot', args.get('firmware')]
+
+    prepare_unattended(params, args, files)
+    prepare_cloud_init(params, args, files)
+
+    logging.debug(params)
+
+    return params
+
+
+def cleanup(files):
+    for file in files:
+        os.unlink(file)
+
+
+def create_vm(args, files):
+    params = prepare_virt_install_params(args, files)
+
+    xml = subprocess.check_output(params)
+
+    if args.get('startVm'):
+        try:
+            assert_vm_exists(args.get('connectionName'), args.get('name'))
+
+            xml = virsh(args.get('connectionName'), "dumpxml", "--inactive", args.get('vmName'))
+        except subprocess.CalledProcessError:
+            logging.info("The VM got deleted while being installed")
+            logging.info(traceback.format_exc())
+            sys.exit(0)
+
+    xml_last_phase = xml.strip().split(b'\n\n')[-1]
+    # Get last step only - virt-install can output 1 or 2 steps
+    inject_metadata(xml_last_phase.decode())
+
+
+def install_vm(args, files):
+    # If the VM already exists and we want to run the installer we need to remove it first
+    virsh(args.get('connectionName'), "destroy", args.get('vmName'), check=False)
```

Comment:
This seems to be the only place you do `check=False`.  At this point I'd lean towards unconditional use of `subprocess.check_output()` and catch (and ignore) the exception in this one place.